compiler/noirc_frontend/src/parser/parser.rs

```diff
@@ -1085,10 +1092,13 @@ where
         .map(UnaryRhs::Cast)
         .labelled(ParsingRuleLabel::Cast);
 
+    // A turbofish operator is optional in a method call to specify generic types
+    let turbofish = primitives::turbofish(parse_type());
```

Comment:
Can you change `atom_or_right_unary` to accept a type parser as a parameter instead? There should be such a parameter already near the expression level. I'm worried not using it here could cause the parser to recur forever in some cases. I think it'd only be when trying to use a method call in a type expression e.g. `let x: Foo<1 + bar.baz::<Field>()>` which we don't allow but it's best to be safe.

Reply:
done